frank/utilities.py

```diff
@@ -272,7 +321,77 @@ def cut_data_by_baseline(u, v, vis, weights, cut_range):
 
     return u_cut, v_cut, vis_cut, weights_cut
 
+def estimate_weights(u, v, V, nbins=300, log=True, use_median=False):
+    r"""
+    Estimate the weights using the variance of the binned visibilities. 
+
+    The estimation is done assuming that the variation in each bin is dominated
+    by the noise. This will be true if:
+        1) The source is axi-symmetric,
+        2) The uv-points have been deprojected,
+    Otherwise the variance may be dominated by real variations in the 
+    visibilities.
+
+    Parameters
+    ----------
+    u, v : array, unit = :math:`\lambda`
+        u and v coordinates of observations (deprojected).
+    V : array, unit = Jy
+        Obsersed visibility. If complex, the weights will be computed from the
+        average of the variance of the real and imaginary components, as in
+        CASA's statwt. Otherwise the variance of the real part is used.
+    nbins : int, default=300
+        Number of bins used.
+    log : bool, default=True
+        If True, the uv bins will be constructed in log space, otherwise linear
+        spaced bins will be used.
+    use_median : bool, default=False
+        If True all of the weights will be set to the median of the variance
+        estimated across the bins. Otherwise, the baseline dependent variance 
+        will be used.
+
+    Returns
+    -------
+    weights : array,
+        Esimtate weight for each uv point.
+
+    Notes
+    -----
+        - This function does not support datasets where the weights may be
+          expected to be different for different baselines, e.g. if the data
+          includes visibilities from telescopes of different sizes
+        - Bins with only one uv point will not have a variance estimate. The
```

Comment:
Would it make more sense to use the variance from the 2 adjacent bins instead of the median across all baselines for bins with only 1 uv point?

Reply:
What if the adjacent bins are empty?

Reply:
The nearest non-empty bin on either side? That still seems a bit more accurate to me than the variance across the full baseline range. I realize in many cases the difference between these 2 results can be small, but I can think of edge cases such as a really skewed baseline distribution where the 'nearest neighbor' estimate would be more accurate.

Reply:
Done

Reply:
The new code can fail if there is only one baseline in the min/max bin